### PR TITLE
Update PSP/RBAC doc for PSS

### PR DESCRIPTION
**Pull request by stone-z**
Towards https://github.com/giantswarm/giantswarm/issues/26092

### What does this PR do?

This removes the PSP section from the RBAC and PSP page and references our separate page on working with policies.

I have not revised the sections on user management, which could still use a refresh from the appropriate team.

### What does it look like?

Mostly textual changes.
Mermaid diagram:
image

### Any background context you can provide?

PSP has been removed from the Kubernetes API, and Giant Swarm chose to use Kyverno instead of the in-tree Pod Security Admission controller as a replacement. This doc needs to change to reflect the updated security controls.

### What is needed from the reviewers?

Opinions: should we change the link for this page? It still references PSP, but this page is mostly about RBAC anyway and is likely linked to in several places.

### Have you maintained the front matter?

Yes!

### CODE REVIEW

File: src/content/getting-started/rbac-and-psp/index.md [hunk 1]
```diff
@@ -484,6 +501,7 @@ roleRef:
   apiGroup: rbac.authorization.k8s.io
 ```
 
+
 ### Running applications that need privileged access
```
Comment (stone-z) [inline src/content/getting-started/rbac-and-psp/index.md:505]: This section is about binding a PSP to an SA to allow a workload to use it -- I'd suggest removing it entirely if that's cool with the reviewers

File: src/content/getting-started/rbac-and-psp/index.md [hunk 2]
```diff
@@ -271,23 +271,40 @@ $ kubectl auth can-i get pods --as-group=system:masters
 yes
 ```
 
-## Pod Security Policies
+## Pod Security Standards (PSS)
 
-A `PodSecurityPolicy` object defines a set of conditions that a pod must run with in order to be accepted into the system. It governs the ability to make requests on a Pod that affect the `SecurityContext` that will be applied to a Pod and container.
+**Note:** Prior to Kubernetes v1.25, pod-level security configuration was controlled by Pod Security Policies (PSP). PSPs were a Kubernetes API resource which allowed cluster administrators to restrict the values a pod could use for certain security-relevant configuration. Due to limitations of the specification, the PSP type was deprecated in v1.21 and removed in v1.25.
```
Comment (pipo02mix) [inline src/content/getting-started/rbac-and-psp/index.md:276]:
```suggestion
__Note__: Prior to Kubernetes v1.25, pod-level security configuration was controlled by Pod Security Policies (PSP). PSPs were a Kubernetes API resource which allowed cluster administrators to restrict the values a pod could use for certain security-relevant configuration. Due to limitations of the specification, the PSP type was deprecated in v1.21 and removed in v1.25.
```

**Review by pipo02mix — APPROVED** [review]
LGTM minor thing, I would add the mermaid diagram after explaining Kyverno to make it more understandable 
good work

File: src/content/getting-started/rbac-and-psp/index.md (hunk 1 above)
Comment (stone-z) [reply src/content/getting-started/rbac-and-psp/index.md]: I've removed this section. Holler at me if someone wants it to stay

**Comment by stone-z** [comment]
>  I would add the mermaid diagram after explaining Kyverno

That makes sense, I moved it into the section about Kyverno

**Review by pipo02mix — APPROVED** [review]

**Review by AndiDog — APPROVED** [review]

**Pull request merged**